spec/index.bs

```diff
@@ -3583,6 +3580,9 @@ interface GPUComputePassEncoder {
     void dispatch(GPUSize32 x, optional GPUSize32 y = 1, optional GPUSize32 z = 1);
     void dispatchIndirect(GPUBuffer indirectBuffer, GPUSize64 indirectOffset);
 
+    void beginPipelineStatisticsQuery(GPUQuerySet querySet, GPUSize32 queryIndex);
```

Comment:
Just a crazy thought: we could just make it a part of the compute pass:
```
dictionary GPUComputePassDescriptor {
    GPUQuerySet? pipelineStatisticsQuerySet;
    optional GPUSize32 pipelineStatisticsQueryIndex = 0;
};
```

That would have a bit less spec text duplication and complexity.

Reply:
To reduce duplication there could alternatively be a `GPUPassEncoderBase` that shares stuff between `GPUComputePassEncoder` and `GPURenderPassEncoder`. In particular we'll want to finer detail control over memory barriers in compute passes eventually, and this would lead to "fatter" compute passes so having a single pipelineStatistics per pass would be too constraining.

Reply:
Another consideration is there is also an occlusionQuerySet in `GPURenderPassDescriptor `, if add a pipelineStatisticsQuerySet, maybe it's a duplication in spec, and there will be timestamp query in the future.